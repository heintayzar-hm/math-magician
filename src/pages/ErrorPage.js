import React from 'react';

// eslint-disable-next-line react/prefer-stateless-function
class ErrorPage extends React.Component {
  render() {
    return (
      <>
        <div>Error</div>
      </>
    );
  }
}

export default ErrorPage;
