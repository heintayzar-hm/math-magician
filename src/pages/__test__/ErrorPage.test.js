import React from 'react';
import { render } from '@testing-library/react';
import ErrorPage from '../ErrorPage';

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => (jest.fn()),
}));

describe('ErrorPage', () => {
  it('renders correctly', () => {
    // Arrange

    // Act
    const { container } = render(<ErrorPage />);

    // Assert
    expect(container.firstChild).toMatchSnapshot();
  });
});
