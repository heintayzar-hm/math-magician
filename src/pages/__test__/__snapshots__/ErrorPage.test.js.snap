// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage renders correctly 1`] = `
<div>
  Error 404
</div>
`;
